cmake/Config.cmake.in

```diff
@@ -10,9 +10,23 @@ set(@PROJECT_NAME@_VERSION @PROJECT_VERSION@)
 # find_dependency(Imath @Imath_VERSION@ HINTS @Imath_DIR@)
 
 set_and_check (@PROJECT_NAME@_DIR "${PACKAGE_PREFIX_DIR}")
+if (WIN32)
+    # For .dll files.
+    set_and_check (@PROJECT_NAME@_BINARY_DIR "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_BINDIR@")
+endif ()
 set_and_check (@PROJECT_NAME@_INCLUDE_DIR "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_INCLUDEDIR@")
 set_and_check (@PROJECT_NAME@_LIBRARY_DIR "${PACKAGE_PREFIX_DIR}/@CMAKE_INSTALL_LIBDIR@")
 
+if (@OPENASSETIO_ENABLE_PYTHON@)
+    set(@PROJECT_NAME@_Python_VERSION "@Python_VERSION@")
+    set(@PROJECT_NAME@_Python_VERSION_MAJOR "@Python_VERSION_MAJOR@")
+    set(@PROJECT_NAME@_Python_VERSION_MINOR "@Python_VERSION_MINOR@")
+    set(@PROJECT_NAME@_Python_VERSION_PATCH "@Python_VERSION_PATCH@")
+    set_and_check (
+        @PROJECT_NAME@_Python_SITELIB
```

Comment:
Can you give an example of where this might be used?

I'm struggling to think in other CMake config files I've seen where this pattern might have been used.

(Thinking along the lines of global-variables-bad in modern CMake)

Reply:
These variables were a bit of a cheeky addition to help the test project https://github.com/feltech/OpenAssetIO-Test/pull/1.  

However, I can see the `Python_SITELIB` variable in particular being useful, to allow downstream projects to reference the installed pure Python sources, either to extract them or to point `PYTHONPATH` at them (which is what the aforementioned test project does).

I guess these could be removed in favour of externally deriving the expected version/location of the Python artifacts, if you think that's a better pattern?

The variables may be global, but at least they're namespaced.

Reply:
I immediately thought I wonder if we can wrap them into more import targets. Precedent here is things like in Boost to just set a preprocessor define, but with a more elegant target name https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L2506-L2507

But... just looked, and saw SITELIB is used as an environment variable, so sadly no good :(

Unless what I've written above has sparked an idea on your side?

Reply:
I was grasping at straws to be honest, in case you saw another way to achieve what you wanted.

Let's stick with vars. Like you say, they're name scoped at least.